From bddfa39c484dc3892064c4262accbc3ad2843c97 Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Mon, 1 Sep 2014 09:25:37 -0400
Subject: [md] raid5: fix new memory-reference bug in alloc_thread_groups

Message-id: <1409563584-9000-36-git-send-email-Jes.Sorensen@redhat.com>
Patchwork-id: 89968
O-Subject: [RHEL7.1 PATCH 35/82] md/raid5: fix new memory-reference bug in alloc_thread_groups.
Bugzilla: 1085530
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: NeilBrown <neilb@suse.de>

In alloc_thread_groups, worker_groups is a pointer to an array,
not an array of pointers.
So
   worker_groups[i]
is wrong.  It should be
   &(*worker_groups)[i]

Found-by: coverity
Fixes: 60aaf9338545
Reported-by: Ben Hutchings <bhutchings@solarflare.com>
Cc: majianpeng <majianpeng@gmail.com>
Signed-off-by: NeilBrown <neilb@suse.de>
(cherry picked from commit 0c775d5208284700de423e6746259da54a42e1f5)
Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/raid5.c b/drivers/md/raid5.c
index d852ae5..a059260 100644
--- a/drivers/md/raid5.c
+++ b/drivers/md/raid5.c
@@ -5473,7 +5473,7 @@ static int alloc_thread_groups(struct r5conf *conf, int cnt,
  for (i = 0; i < *group_cnt; i++) {
   struct r5worker_group *group;
 
-  group = worker_groups[i];
+  group = &(*worker_groups)[i];
   INIT_LIST_HEAD(&group->handle_list);
   group->conf = conf;
   group->workers = workers + i * cnt;
-- 
1.7.1